doc/dev/test_proxy_migration_guide.md

```diff
@@ -0,0 +1,174 @@
+container, providing the full path to the root of your local `azure-sdk-for-python` repo in place of `{path}`:
+
+```cmd
+docker container create -v C:\\{path}\\azure-sdk-for-python:/etc/testproxy -p 5001:5001 -p 5000:5000 --name ambitious_azsdk_test_proxy azsdkengsys.azurecr.io/engsys/testproxy-lin
+```
+
+Some tests require an SSL connection to work, so the Docker image used for the container has a certificate imported
+that you need to trust on your machine. Instructions on how to do so can be found [here][proxy_cert_docs].
+
+### Start the proxy server
+
+Run the following command to start the container whenever you want to make the test proxy available:
+
+```cmd
+docker container start ambitious_azsdk_test_proxy
+```
+
+Note that the proxy is available as long as the container is running. In other words, you don't need to start and
+stop the container for each test run or between tests for different SDKs. You can run the above command in the morning
+and just stop the container whenever you'd like. In the future, the proxy container will be set up and started
+automatically when tests are run, and starting it manually will be optional.
+
+For more details on proxy startup, please refer to the [proxy documentation][detailed_docs].
+
+### Record or play back tests
+
+Configuring live and playback tests is done with the `AZURE_TEST_RUN_LIVE` environment variable. When this variable is
+set to "true" or "yes", live tests will run and produce recordings. When this variable is set to "false" or "no", or
+not set at all, tests will run in playback mode and attempt to match existing recordings.
+
+Recordings for a given package will end up in that package's `/tests/recordings` directory, just like they currently
+do.
+
+> **Note:** at this time, support for configuring live or playback tests with a `testsettings_local.cfg` file has been
+> deprecated in favor of using just `AZURE_TEST_RUN_LIVE`.
+
+### Register sanitizers
+
+Since the test proxy doesn't use [`vcrpy`][vcrpy], tests don't use a scrubber to sanitize values in recordings.
+Instead, sanitizers (as well as matchers and transforms) can be registered on the proxy as detailed in
+[this][sanitizers] section of the proxy documentation. At the time of writing, sanitizers can be registered via
+the `add_sanitizer` method in `devtools_testutils`. For example, at the start of a test file, you can set up a URI
+sanitizer for all tests by doing something like the following:
+
+```python
+from devtools_testutils import add_sanitizer, AzureRecordedTestCase, ProxyRecordingSanitizer
+
+add_sanitizer(ProxyRecordingSanitizer.URI, value="fakeendpoint")
+
+class TestExample(AzureRecordedTestCase):
+    ...
+```
+
+`add_sanitizer` accepts a sanitizer type from the ProxyRecordingSanitizer enum as a required parameter. A regular
+expression to match and a value to use in recordings can be optionally provided as `regex` and `value` arguments,
```

Comment:
Right now we only provide `regex` and `value` right.

All the regex sanitizers also allow a 3rd argument `groupForReplace`. Maybe we swap `add_sanitizer` to variable number of arguments, throwing if we don't get the expected ones. We could generate some nice errors fairly simply.